runelite-client/src/main/java/net/runelite/client/plugins/farmingtracker/FarmingTrackerPlugin.java

```diff
@@ -0,0 +1,181 @@
+ *  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.farmingtracker;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import java.awt.image.BufferedImage;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import javax.imageio.ImageIO;
+import javax.inject.Inject;
+import javax.swing.SwingUtilities;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.Varbits;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.UsernameChanged;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.task.Schedule;
+import net.runelite.client.ui.NavigationButton;
+import net.runelite.client.ui.PluginToolbar;
+
+@PluginDescriptor(
+	name = "Farming Tracker"
+)
+@Slf4j
+public class FarmingTrackerPlugin extends Plugin
+{
+	@Inject
+	private PluginToolbar pluginToolbar;
+
+	@Inject
+	private ConfigManager configManager;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private FarmingWorld farmingWorld;
+
+	private FarmingTrackerPanel panel;
+
+	private NavigationButton navButton;
+
+	private WorldPoint lastTickLoc;
+
+	@Provides
+	FarmingTrackerConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(FarmingTrackerConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		BufferedImage icon;
+		synchronized (ImageIO.class)
+		{
+			icon = ImageIO.read(getClass().getResourceAsStream("farming.png"));
+		}
+
+		panel = injector.getInstance(FarmingTrackerPanel.class);
+		panel.init();
+
+		navButton = NavigationButton.builder()
+			.name("Farming Tracker")
+			.icon(icon)
+			.panel(panel)
+			.build();
+
+		pluginToolbar.addNavigation(navButton);
+
+		updatePanel();
+	}
+
+	@Subscribe
+	public void onUsernameChanged(UsernameChanged e)
+	{
+		updatePanel();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		pluginToolbar.removeNavigation(navButton);
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick t)
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			lastTickLoc = null;
+			return;
+		}
+
+		WorldPoint loc = lastTickLoc;
+		lastTickLoc = client.getLocalPlayer().getWorldLocation();
+		if (loc == null)
+		{
+			return;
+		}
+
+		FarmingRegion region = farmingWorld.getRegions().get(loc.getRegionID());
+		if (region != null)
+		{
+			// Write config with new varbits
+			// farmingTracker.<login-username>.<regionID>.<VarbitID>=<varbitValue>:<unix time>
+			String group = FarmingTrackerConfig.KEY_NAME + "." + client.getUsername() + "." + region.getRegionID();
```

Comment:
This seems like workaround until proper user-based settings are added? I don't think this should be done and just leave it be just like SlayerPlugin works atm

Reply:
I've had multiple people ask me if this was going to work for multiple accounts, and I don't see proper per-user settings coming for at least a couple weeks, so I would like to keep this behavior.